### PR TITLE
feat: mocked QuizRepository and update end2end test Accordingly i.e solve the quiz

### PR DESCRIPTION
**Pull request by SaFouNaldo**
- This PR introduces a MockQuizRepository implementation and corresponding unit tests to enhance the end-to-end (E2E) testing capabilities for the HomeScreen feature. Previously, the E2E test assumed the quiz would automatically be solved without simulating user actions, such as selecting an answer and clicking the submit button. By mocking the QuizRepository, we enable controlled, predictable behavior during tests, allowing for accurate simulation of user interactions.

## Changes:
1. MockQuizRepository Implementation:

- Provides a simple mocked quiz question with predefined data ("apple" as the correct answer with signs and confusers).
Includes methods to simulate success or failure scenarios, making it easier to test edge cases.
2. MockQuizRepositoryTest Unit Tests:

- Verifies that the repository correctly returns mocked quiz data in success scenarios.
- Ensures appropriate failure callbacks are triggered in failure scenarios.
- Validates the initialization behavior when simulating success and failure.
3. Enhanced E2E Test Integration:

- The MockQuizRepository can now be injected via DependencyProvider to control test data.
- Simulates user behavior, including selecting answers and submitting them, making the test flow more realistic and comprehensive.
## Motivation:
These changes solve the problem where the E2E test lacked the ability to simulate user behavior during the quiz flow. By introducing a mock repository, we improve test reliability, maintainability, and realism, ensuring better coverage and easier debugging.

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/signify/end2end/HomeScreenFeaturesEnd2endTest.kt [hunk 1]
```diff
@@ -103,6 +103,11 @@ class HomeScreenFeaturesEnd2endTest {
     composeTestRule.onNodeWithTag("HomeScreen").assertIsDisplayed()
     composeTestRule.onNodeWithTag("QuizButton").performClick()
     composeTestRule.onNodeWithTag("QuizTitle").assertIsDisplayed()
+
+    // Simulate choosing the correct solution based on mocked data
```
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/end2end/HomeScreenFeaturesEnd2endTest.kt:107]: Since quizzes are not tracked and are considered a free game in a sense, it’s a good idea to test them while interacting with the home screen.

File: app/src/main/java/com/github/se/signify/model/home/quiz/MockQuizRepository.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/model/home/quiz/MockQuizRepository.kt:1]: A well-structured mock repository with clear methods to simulate success and failure scenarios, reusable failure handling, and predefined mock data for consistent testing.

File: app/src/androidTest/java/com/github/se/signify/model/home/quiz/MockQuizRepositoryTest.kt
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/model/home/quiz/MockQuizRepositoryTest.kt:1]: This test class is well-structured and provides comprehensive coverage of the MockQuizRepository. It effectively tests both success and failure scenarios, ensuring that the repository behaves as expected in different states.

**Review by ziyadbagh — APPROVED** [review]
Approved !!
